ft: ZENKO-1019 delete scheduled resume

philipyoo opened the pull request:
**Looking to merge this for pre-GA build**

- Add a DELETE route to delete/cancel a scheduled resume for given site(s)
- Small refactor to reuse a test helper method for both POST and DELETE requests

philipyoo committed:
ft: ZENKO-1019 add delete for scheduled resume

Add a DELETE route to remove a schedule resume

bert-e commented:
# Hello philipyoo,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get
information on this process.





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @vrancurel

* @rahulreddy

* @bennettbuchanan

* @jonathan-gramain

philipyoo commented:
@bert-e wait

rahulreddy commented:
@RachedBenMustapha  This may be of interest to you. Did you already implement a cancel by sending an `{}` body from Orbit?

rachedbenmustapha commented:
@alexanderchan-scality can you confirm that the immediate resume is using an empty object as a body placeholder already?

alexanderchan-scality commented:
@RachedBenMustapha Yes, the immediate resume/pause requests have empty bodies. As for @rahulreddy 's question about `{}` for the schedule requests, an empty body will default to scheduling a resume in 6 hours, so the empty will not unset the scheduled times.

JianqinWang commented on lib/api/BackbeatAPI.js:
```diff
@@ -736,6 +736,9 @@ class BackbeatAPI {
         }
         try {
             reqBody = JSON.parse(body);
+            if (reqBody.cancel) {
```
If a user set this to 'false', wouldn't this skip?
In the documentation it says:
"Providing a POST request body object with a `cancel` key with any value will remove any scheduled resumes for the given destination replication endpoint."
We probably shouldn't give users this much freedom of usage + interpretation.

philipyoo replied:
ah good catch! I think I'm going to change this so rather than any value, I expect user to send `true`. Seems sort of weird to just accept any value from a ux perspective

bennettbuchanan commented on docs/crr-pause-resume.md:
````diff
@@ -141,6 +142,20 @@ is resumed, it again resumes consuming entries from its last offset.
     }
     ```
 
+    Providing a POST request body object with a `cancel` key with any value
````
Looks like it has to be a truthy value, rather than any value.

bennettbuchanan commented on lib/api/BackbeatAPI.js:
```diff
@@ -736,6 +736,17 @@ class BackbeatAPI {
+                if (reqBody.cancel !== true) {
+                    return {
+                        error: errors.MalformedPOSTrequest.customizeDescription(
+                            `${msg}: cancel key found with incorrect value`),
```
I think it would be worthwhile to include in the message that the `cancel` property key must be `true`.

philipyoo commented:
Sorry instead of cancel via POST, should be a DELETE request with no body

rahulreddy commented:
It's good to be inline with the REST API semantics 💯

philipyoo committed:
rf: rename and reuse http request method

Rename and reuse the http request helper method for use
with both POST and DELETE requests

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.0`



The following branches will **NOT** be impacted:

* `development/7.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of Release Engineering will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of Release Engineering now.

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.0`



The following branches have **NOT** changed:

* `development/7.4`



Please check the status of the associated issue ZENKO-1019.

Goodbye philipyoo.
